stellargraph/layer/watch_your_step.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer, Embedding, Input, Lambda, Concatenate, Dense
+from tensorflow.keras import backend as K
+import warnings
+from ..mapper.adjacency_generators import AdjacencyPowerGenerator
+from ..core.experimental import experimental
+
+
+@experimental(reason="lack of unit tests")
+class AttentiveWalk(Layer):
+    """
+    This implements the graph attention as in Watch Your Step: Learning Node Embeddings via Graph Attention
+    https://arxiv.org/pdf/1710.09599.pdf.
+
+    Args:
+        walk_length (int): the length of the random walks. Equivalent to the number of adjacency powers used. Defaults
+            to `10` as this value was found to perform well by the authors of the paper.
+        attention_initializer (str or func, optional): The initialiser to use for the attention weights.
+        attention_regularizer (str or func, optional): The regulariser to use for the attention weights.
+        attention_constraint (str or func, optional): The constraint to use for the attention weights.
+        input_dim (tuple of ints, optional): The shape of the input to the layer.
+    """
+
+    def __init__(
+        self,
+        walk_length=10,
+        attention_initializer="glorot_uniform",
+        attention_regularizer=None,
+        attention_constraint=None,
+        input_dim=None,
+        **kwargs
+    ):
+
+        if (not "input_shape" in kwargs) and (not input_dim is None):
```

Comment:
I think it's slightly more Pythonic to be more English-y:

```suggestion
        if "input_shape" not in kwargs and input_dim is not None:
```